test/galore/galore_memory_analysis_utils.py

```diff
@@ -0,0 +1,72 @@
+from functools import partial
```

Comment:
if possible I'd rather we not have notebooks merged in, just turn the code into a python file and let's check that in